src/openms/source/METADATA/PeptideHit.cpp

```diff
@@ -198,6 +198,11 @@ namespace OpenMS
     sequence_ = sequence;
   }
 
+  void PeptideHit::setSequence(AASequence&& sequence)
+  {
+    sequence_ = std::move(sequence);
```

Comment:
could this assignment potentially be marked as throwing? This would probably call the move operator (and not the move constructor). See my comments below.

Reply:
hmm it could since its not labeled `noexept` yes, but on the other hand its not used in a `std::vector`, right? I am not sure I get your point?

Reply:
ok I think I get it now. Even if the elements of the underlying vector  would throw then this std::move would still perform a cheap e.g., std::swap

Reply:
still not sure I understand what this has to do with throwing, the only thing this will do is to call

```
AASequence& operator=(AASequence&&);
```

which is default constructed (e.g. `AASequence& operator=(AASequence&&); = default`) and most likely will do the following (this seems most likely):

```    
peptide_ == std::move(rhs.peptide_);
n_term_mod_ == std::move(rhs.n_term_mod_);
c_term_mod_ == std::move(rhs.n_term_mod_);
``` 

which then will call the implementation of `std::vector<const Residue*>& operator=(std::vector<const Residue*>&&);` which is the one from `std::vector` - again it does not matter whether its noexcept or not, this will be called. Now there is an interesting question of whether it should be 

```
std::vector<T>& operator=(std::vector<T>&&);
--- or
std::vector<T>& operator=(std::vector<T>&&) noexcept;
```

see here, https://stackoverflow.com/questions/12332772/why-arent-container-move-assignment-operators-noexcept -- but again irrelevant for our case here. So what will  `std::vector<const Residue*>& operator=(std::vector<const Residue*>&&);` probably do? (Again its not `noexcept` so it does not even guarantee anything) From my understanding, it will simply swap the ptrs to memory of `*this` and `rhs`.

I think you may be confusing the move assignment and move constructors of the `std::vector` and their `noexcept`ness with the question of whether the elements of `std::vector<T>` have noexcept move assignment and move constructors (e.g. `T` has noexcept move assignment and move constructors ). This is relevant when the vector resizes.